src/java.desktop/share/native/common/java2d/opengl/OGLBlitLoops.c

```diff
@@ -321,23 +321,26 @@ OGLBlitToSurfaceViaTexture(OGLContext *oglc, SurfaceDataRasInfo *srcInfo,
                 GLvoid *pSrc = PtrCoord(srcInfo->rasBase,
                                         sx, srcInfo->pixelStride,
                                         sy, srcInfo->scanStride);
-                if (slowPath) {
-                    jint tmph = sh;
-                    while (tmph > 0) {
-                        j2d_glTexSubImage2D(GL_TEXTURE_2D, 0,
+                if (pf != NULL) {
```

Comment:
I don't think any of these changes in this file are necessary.
pf is null IFF sw_surface == JNI_FALSE 
and iin that case, we never end the loop where pf is needed.
Put another way, if you look at the call sites one 
is 
pf != null, srcOps == NULL, sw_surface == FALSE
the other is
pf == null, srcOps != NULL, sw_surface == TRUE

and you can see that srcOps would have the same theoretical issue except that it would be a bug to call it with the wrong value of sw_surface.
So revert all changes in this file.

Reply:
Done.